rosidl_cmake/cmake/rosidl_generate_interfaces.cmake

```diff
@@ -137,21 +137,48 @@ macro(rosidl_generate_interfaces target)
   # afterwards all remaining interface files are .idl files
   list(APPEND _idl_tuples ${_idl_adapter_tuples})
 
-  # to generate action interfaces, we need to depend on "action_msgs"
+  # Check for any action or service interfaces
+  # Which have implicit dependencies that need to be found
   foreach(_tuple ${_interface_tuples})
+    # We use the parent directory name to identify if the interface is an action or service
     string(REGEX REPLACE ".*:([^:]*)$" "\\1" _tuple_file "${_tuple}")
     get_filename_component(_parent_dir "${_tuple_file}" DIRECTORY)
```

Comment:
Turns out this logic for using the parent directory is quite brittle :roll_eyes: 

With the introduction of the `service_msgs` dependency, this PR has caused a regression in `rqt_py_common`, which doesn't try to `find_package(service_msgs)` because the parent directory is `test/srv` (not `srv`): https://github.com/ros-visualization/rqt/blob/d0234e0c3a96e6815e0a650e48fae97c7d877f92/rqt_py_common/CMakeLists.txt#L35

Now might be a good time to rethink this heuristic.

Reply:
See [4409291](https://github.com/ros2/rosidl/pull/700/commits/4409291176aada4e79ac249e60f6a386769e8418)
It proposes only relying on the immediate parent directory name for detecting action and service interfaces.

Reply:
that seems fine to me